jupyter/nmm-game-utils.ipynb

```diff
@@ -139,7 +191,73 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Player Phase"
+    "## Funktionen für den Spieler"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### opponent\n",
+    "\n",
+    "Die Funktion `opponent` nimmt einen Spieler und gibt den Gegenspieler zurück. Die Funktion hat ein Argument:\n",
+    "\n",
+    "- `p` ist der aktuelle Spieler.\n",
+    "\n",
+    "Da Mühle ein Zwei-Personen-Spiel ist, gibt es für die Funktion nur zwei Fälle:\n",
+    "\n",
+    "- bei einem weißen Spieler `'w'` wird der Gegenspieler schwarz `'b'` zurückgegeben,\n",
+    "- ansonsten wird weiß `'w'` als Gegenspieler zurückgegeben."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def opponent(p):\n",
+    "    return 'b' if p == 'w' else 'w'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### getPlayerAt\n",
+    "\n",
+    "Die Funktion `getPlayerAt` gibt den Spieler an der gegeben Koordinate des Spielbrettes zurück. Die Funktion hat zwei Argumente:\n",
+    "\n",
+    "- `board` ist ein Spielbrett;\n",
+    "- `coord` ist eine Koordinate, die überprüft werden soll. \n",
+    "\n",
+    "Die Funktion gibt einen Spieler zurück. Falls an dieser Koordinate sich kein Spielerstein befinden sollte, wird entsprechend `' '` zurückgegeben."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Returns the player on the given coord\n",
+    "def getPlayerAt(board, coord):\n",
+    "    (r, c) = coord\n",
+    "    return board[r][c]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### playerPhase\n",
+    "\n",
+    "Die Funktion `playerPhase` berechnet für einen gegebenen Zustand und einen Spieler die aktuelle Phase des Spielers. Die Funktion hat zwei Argumente:\n",
+    "\n",
+    "- `s` ist der aktuelle Zustand eines Spiels,\n",
+    "- `p` ist der aktuelle Spieler.\n",
+    "\n",
+    "Die Funktion überprüft mit den beiden Hilfsfunktionen `hasPlaceableStones` und `isAllowedToJump` die Spielerphase und gibt diese als Ganzzahl zurück."
```

Comment:
Hier sollte man noch kurz erklären, dass `1` die _placing_ Phase, etc ist. Und man kann auch noch kurz sagen, wann welche Phase warum eintritt.

Reply:
Das wird doch in der game implementation

Dachte das ist dann unnötig es nochmal zu machen ^^

Reply:
Hmm nagut

Reply:
Hab es jetzt doch hinzugefügt:

Die Funktion überprüft mit den beiden Hilfsfunktionen `hasPlaceableStones` und `isAllowedToJump` die Spielerphase und gibt diese als Ganzzahl zurück:

- `1` für die _placing_ Phase g.d.w. der Spieler noch Steine auf dem Stapel hat (`hasPlaceableStones`);
- `2` für die _moving_ Phase g.d.w. der Spieler nicht in Phase 1 oder 3 ist;
- `3` für die _flying_ Phase g.d.w. der Spieler springen darf (`isAllowedToJump`).